lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp

```diff
@@ -64,43 +64,47 @@ void GPUXToSPIRVPass::runOnOperation() {
     gpuModules.push_back(builder.clone(*moduleOp.getOperation()));
   });
 
-  // Map MemRef memory space to SPIR-V storage class first if requested.
-  if (mapMemorySpace) {
-    std::unique_ptr<mlir::ConversionTarget> target =
-        mlir::spirv::getMemorySpaceToStorageClassTarget(*context);
-    mlir::spirv::MemorySpaceToStorageClassMap memorySpaceMap =
-        mlir::spirv::mapMemorySpaceToVulkanStorageClass;
-    mlir::spirv::MemorySpaceToStorageClassConverter converter(memorySpaceMap);
+  for (auto gpuModule : gpuModules) {
```

Comment:
so this means upstream is not supporting iterating over all gpu modules?

Reply:
upstream does it as well ..here https://mlir.llvm.org/doxygen/GPUToSPIRVPass_8cpp_source.html ..line 68

Reply:
Yes, I see it now, when I added this pass, upstream did not have this covered. Looks good.

Reply:
I did this change both in `main` and in upstream, to support attaching different TargetEnv's to different gpu modules https://reviews.llvm.org/D135907